### PR TITLE
Updating and replacing OGW troubleshooting chap

### LINKED ISSUE #562: Update ogw content troubleshooting guide
**Link To Section**
https://gist.github.com/theanalyst/c1c9b03665a7d31775264af20e1a6fb4

**Summary**
Update https://susedoc.github.io/doc-ses/master/html/ses-troubleshooting/bp-troubleshooting-objectgateway.html with the content @theanalyst has written above in the gist

### PR DESCRIPTION
**Pull request by asettle**
Closes #562

### CODE REVIEW

File: xml/bp_troubleshooting_objectgateway.xml [hunk 1]
```diff
@@ -13,143 +13,141 @@
...
+     messenger debug levels can be raised (off by default). This is controlled
+     via <literal>debug ms</literal>. Usually a level of 1 is sufficient to
+     gather enough information. For injecting arguments in a running process:
    </para>
+<screen>
+&prompt.cephuser;ceph-daemon <replaceable>client.rgw</replaceable> config set debug_rgw 20
+</screen>
    <para>
-     Check a cluster's status:
+     Or, alternatively, using <command>ceph-daemon</command>:
```
Comment (theanalyst) [inline xml/bp_troubleshooting_objectgateway.xml:54]: move the using `ceph-daemon` to the previous line (before the screen), Let this one be just Or, alternatively, using the `ceph` cli

File: xml/bp_troubleshooting_objectgateway.xml [hunk 2]
```diff
@@ -160,124 +158,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
...
+    it might indicate one of the following:
+  </para>
+  <sect2 xml:id="resharding-issues">
+    <title>Resharding Issues</title>
+    <para>
+      If the large omap object name happens to be in the
+      <literal><replaceable>zone</replaceable>.rgw.buckets.index</literal> pool
+      this means that a bucket has more than 200K keys, dynamic resharding is
+      set on by default in single site clusters which would automatically have
+      resharded the bucket, for multi site clusters however this is not supported
```
Comment (tbazant) [inline xml/bp_troubleshooting_objectgateway.xml:183]:
```suggestion
      resharded the bucket. For multi-site clusters, however, this is not supported
```

File: xml/bp_troubleshooting_objectgateway.xml [hunk 3]
```diff
@@ -160,124 +158,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
...
+      and would have to be manually done <xref linkend="ogw-bucket-reshard"/>. Looking at the
+      bucket stats would reveal the total object count and the number of shards.
```
Comment (tbazant) [inline xml/bp_troubleshooting_objectgateway.xml:185]:
```suggestion
      bucket statistics would reveal the total object count and the number of shards.
```

File: xml/bp_troubleshooting_objectgateway.xml [hunk 4]
```diff
@@ -160,124 +158,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
...
+      This is helpful to understand if the bucket has already been resharded,
+      in which case the warning is just because the deep scrub happened to run
+      before the reshard process and would be cleared in the next deep scrub.
+    </para>
+  </sect2>
+  <sect2 xml:id="usage-statistics">
+    <title>Usage Statistics</title>
+    <para>
+      Usage stats are also stored as omap keys in the <literal><replaceable>zone</replaceable>.rgw.log</literal>
```
Comment (tbazant) [inline xml/bp_troubleshooting_objectgateway.xml:194]:
```suggestion
      Usage statistics are also stored as OMAP keys in the <literal><replaceable>zone</replaceable>.rgw.log</literal>
```

File: xml/bp_troubleshooting_objectgateway.xml [hunk 5]
```diff
@@ -160,124 +158,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
...
+      pool, however these are not automatically trimmed, so a manual trimming
```
Comment (tbazant) [inline xml/bp_troubleshooting_objectgateway.xml:195]:
```suggestion
      pool. However, these are not automatically trimmed, therefore a manual trimming
```

File: xml/bp_troubleshooting_objectgateway.xml [hunk 6]
```diff
@@ -160,124 +158,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
...
+      of usage stats would have to be done. For example:
```
Comment (tbazant) [inline xml/bp_troubleshooting_objectgateway.xml:196]:
```suggestion
      of usage statistics would have to be done. For example:
```

File: xml/bp_troubleshooting_objectgateway.xml [hunk 7]
```diff
@@ -160,124 +158,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
...
+    </para>
 <screen>
-&prompt.cephuser.ogw;radosgw-admin zone get --rgw-zone=default | grep default.rgw.meta
-[should not match any strings]
+&prompt.smaster;radosgw-admin usage trim [--uid=<replaceable>user-id</replaceable>] --start-date=2019-01-01 --end-date=2019-03-03
```
Comment (tbazant) [inline xml/bp_troubleshooting_objectgateway.xml:199]: 'smaster' prompt? or is it going to be run on salt master, or better on the admin node?

**Review by tbazant — CHANGES_REQUESTED** [review]
well,done, few points to clarify :-)

File: xml/bp_troubleshooting_objectgateway.xml (hunk 7 above)
Comment (asettle) [reply xml/bp_troubleshooting_objectgateway.xml:199]: Hmm probably admin node, sorry. I'll update.

Comment (asettle) [inline xml/bp_troubleshooting_objectgateway.xml:199]:
```suggestion
&prompt.root;radosgw-admin usage trim [--uid=<replaceable>user-id</replaceable>] --start-date=2019-01-01 --end-date=2019-03-03

File: xml/bp_troubleshooting_objectgateway.xml [hunk 8]
```diff
@@ -13,143 +13,141 @@
...
-&prompt.cephuser;ceph status
+&prompt.cephuser;ceph config set <replaceable>client.rgw-name</replaceable> debug_rgw 20
 </screen>
-    <para>
-      &ceph; will print the cluster status. For more information on monitoring
-      the cluster, see <xref linkend="monitor-monstatus"/>.
-    </para>
+   <para>
+     For persistent changes or a non running process it is best to set these
+     lines in the <filename>ceph.conf</filename> file under the <literal>client.<replaceable>rgw-name</replaceable></literal>
```
[the remark this answers is not in the document]
Comment (theanalyst) [reply xml/bp_troubleshooting_objectgateway.xml:61]: Ok we can move this line above, so we could for persistent changes either ceph.conf or the `ceph config set` works

The ceph-daemon thing is temporary

File: xml/bp_troubleshooting_objectgateway.xml [hunk 9]
```diff
@@ -160,124 +161,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
...
-     </para>
-     <para>
-       Users should check zone configuration before proceeding any cleanup procedures:
-     </para>
+ <sect1 xml:id="large-omap-issues">
+  <title>Large OMAP Issues</title>
+  <para>
+    If you receive a cluster warning for a large OMAP issue, it means that
+    some object has exceeded one of the osd deep scrub large OMAP object *
+    threshold, usually set to 200K keys and 1G of storage. The relevant pool or PG
```
Comment (asettle) [inline xml/bp_troubleshooting_objectgateway.xml:174]:
```suggestion
    threshold, usually set to 200,000 keys and 1 GiB of storage. The relevant pool or PG

File: xml/bp_troubleshooting_objectgateway.xml [hunk 10]
```diff
@@ -160,124 +161,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
...
+    involved can be found by grepping for large OMAP object as mentioned in
+    the warning message. Depending on the pool and the object key involved
+    it might indicate one of the following:
+  </para>
+  <sect2 xml:id="resharding-issues">
+    <title>Resharding Issues</title>
+    <para>
+      If the large OMAP object name happens to be in the
+      <literal><replaceable>zone</replaceable>.rgw.buckets.index</literal> pool
+      this means that a bucket has more than 200K keys, dynamic resharding is
```
Comment (asettle) [inline xml/bp_troubleshooting_objectgateway.xml:184]:
```suggestion
      this means that a bucket has more than 200,000 keys, dynamic resharding is

File: xml/bp_troubleshooting_objectgateway.xml [hunk 11]
```diff
@@ -160,124 +161,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
...
-       other critical metadata.
-     </para>
-     <para>
-       Users should check zone configuration before proceeding any cleanup procedures:
-     </para>
+ <sect1 xml:id="large-omap-issues">
+  <title>Large OMAP Issues</title>
+  <para>
+    If you receive a cluster warning for a large OMAP issue, it means that
+    some object has exceeded one of the osd deep scrub large OMAP object *
```
Comment (asettle) [inline xml/bp_troubleshooting_objectgateway.xml:173]:
```suggestion
    an object has exceeded one of the OSD deep scrub large OMAP object

File: xml/bp_troubleshooting_objectgateway.xml [hunk 12]
```diff
@@ -13,143 +13,144 @@
...
+     point your browser or client at the &ogw; endpoint. This should return
+     an empty bucket results for the anonymous user:
    </para>
 <screen>
-/etc/init.d/radosgw start -v
+&prompt.smaster;curl -I
 </screen>
    <para>
-     Or:
+     If the result returns a 405 status, this indicates MethodNotAllowed.
```
Comment (asettle) [inline xml/bp_troubleshooting_objectgateway.xml:27]:
```suggestion
     If the result returns a 405 status, this indicates <literal>MethodNotAllowed</literal>.

Comment (theanalyst) [reply xml/bp_troubleshooting_objectgateway.xml:27]: similarly `NoSuchBucket`

**Review by theanalyst — APPROVED** [review]

File: xml/bp_troubleshooting_objectgateway.xml [hunk 13]
```diff
@@ -170,8 +170,8 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
   <title>Large OMAP Issues</title>
   <para>
     If you receive a cluster warning for a large OMAP issue, it means that
-    some object has exceeded one of the osd deep scrub large OMAP object *
-    threshold, usually set to 200K keys and 1G of storage. The relevant pool or PG
+    an object has exceeded one of the OSD deep scrub large OMAP object
+    threshold, usually set to 200,000 keys and 1 GiB of storage. The relevant pool or PG
```
Comment (theanalyst) [inline xml/bp_troubleshooting_objectgateway.xml:174]: thresholds?

File: xml/bp_troubleshooting_objectgateway.xml [hunk 14]
```diff
@@ -13,143 +13,144 @@
    <dm:release>SES 7</dm:release>
   </dm:docmanager>
  </info>
- <sect1 xml:id="bp-troubleshooting-gateway">
-   <title>The Gateway Will Not Start</title>
+ <sect1 xml:id="troubleshooting-gateway-healthcheck">
+   <title>Basic healthcheck</title>
```
Comment (tbazant) [inline xml/bp_troubleshooting_objectgateway.xml:17]:
```suggestion
   <title>Basic Health Check</title>
```

File: xml/bp_troubleshooting_objectgateway.xml [hunk 15]
```diff
@@ -13,143 +13,144 @@
...
-     proper location.
+     Usually a restarted gateway (automatic under &systemd;) should
+     restore the service. This can be achieved via:
    </para>
+<screen>
+&prompt.cephuser;ceph orch daemon restart <replaceable>rgw-daemon-name</replaceable>
+</screen>
    <para>
-     Check a cluster's status:
+     For bumping up the log level, under the <literal>client.<replaceable>RGW_NAME</replaceable></literal>
```
Comment (tbazant) [inline xml/bp_troubleshooting_objectgateway.xml:43]: "bump up"? meaning "increase"?

**Review by tbazant — CHANGES_REQUESTED** [review]
minor stuff

**Review by tbazant — APPROVED** [review]
lgtm

**Pull request merged**